Simplenote/SPAppDelegate+Extensions.swift

```diff
@@ -408,16 +409,25 @@ extension SPAppDelegate {
 //
 extension SPAppDelegate {
     @objc
-    func authenticateSimperiumIfNeeded() {
-        guard let deletionController = accountDeletionController else {
+    func authenticateSimperiumIfAccountDeletionRequested() {
+        guard let deletionController = accountDeletionController,
+              deletionController.hasValidDeletionRequest else {
             return
         }
 
-        if deletionController.deletionTokenHasExpired {
+        simperium.authenticateIfNecessary()
+    }
+
+    @objc
+    func logOutIfAccountDeletionRequested() {
+        guard let deletionController = accountDeletionController,
+              deletionController.hasValidDeletionRequest,
+              let user = simperium.user,
+              user.authenticated() else {
```

Comment:
`user.authenticated()` will be false because we come here when user token is invalid, which means user is not authenticated. 
We don't need this condition here and can safely remove it.

Reply:
After I removed `user.authenticated()` and tested the account deletion flow, I found another issue.

Calling `[self.simperium signOutAndRemoveLocalData:YES completion:^{}];` doesn't do anything because the first line there is checking if user is authenticated. It will return `false` in our case because the token is not valid and `singOut` won't do anything because of that. This means that all the data is still stored in the core data and we failed to remove it.

https://github.com/Simperium/simperium-ios/blob/develop/Simperium/Simperium.m#L589
```
- (void)signOutAndRemoveLocalData:(BOOL)remove completion:(SimperiumSignoutCompletion)completion {
    
    // Don't proceed, if the user isn't logged in
    if (!self.user.authenticated) {
        return;
    }

   ...
```

The issue here is that simperium checks for `self.user.authenticated` which can be false, but we will still have the user object and credentials stored in the keychain and on the next app launch Simperium will try to use the token from the keychain. The check is quite misleading.

I suggest that we change ` if (!self.user.authenticated)` to ` if (!self.user)` or even fully remove the line. This should allow us to fully clean the state even if the user is not authenticated.

@jleandroperez WDYT?